added functionality for authorization, authentication with regular an…

dmytrosmetanin opened the pull request:
…d refresh tokens, e-mail verification of user

dmytrosmetanin committed:
added functionality for authorization, authentication with regular and refresh tokens, e-mail verification of user

vvelizariy commented on codeArena/codeArena/settings.py:
```diff
@@ -12,6 +12,7 @@
 
 from pathlib import Path
 from decouple import config
+from datetime import timedelta
```
Alpabetical sorting
from datetime import timedelta
from decouple import config
from pathlib import Path

vvelizariy commented on codeArena/codeArena/settings.py:
```diff
@@ -163,18 +168,103 @@
+    'social_core.pipeline.user.user_details',
+)
+
+# SOCIAL_AUTH_FACEBOOK_KEY = os.environ.get("FACEBOOK_KEY", "")
```
Commented lines should not be present

vvelizariy commented on codeArena/user/models.py:
```diff
@@ -23,51 +25,54 @@ def _create_user(self, email, password, **extra_fields):
+            return user
+          
         except:
-            raise
```
No sense in try except block, if you doesnt handle error, it will be raised automatically

vvelizariy commented on codeArena/user/models.py:
```diff
@@ -23,51 +25,54 @@ def _create_user(self, email, password, **extra_fields):
-        extra_fields.setdefault('is_staff', True)
-        extra_fields.setdefault('is_superuser', True)
-        return self._create_user(email, password=password, **extra_fields)
+    # def create_superuser(self, email, password, **extra_fields):
```
commented lines

vvelizariy commented on codeArena/user/models.py:
```diff
@@ -23,51 +25,54 @@ def _create_user(self, email, password, **extra_fields):
-    nickname = models.CharField(max_length=70, unique=True)
-    first_name = models.CharField(max_length=70)
-    last_name = models.CharField(max_length=70)
+    # nickname = models.CharField(max_length=70, unique=True)
```
same, comment

vvelizariy commented on codeArena/user/models.py:
```diff
@@ -23,51 +25,54 @@ def _create_user(self, email, password, **extra_fields):
-        super(User, self).save(*args, **kwargs)
-        return self
+    # def save(self, *args, **kwargs):
+    #     super(User, self).save(*args, **kwargs)
```
comment

vvelizariy commented on codeArena/user/serializers.py:
```diff
@@ -9,58 +9,54 @@ class UserSerializer(serializers.ModelSerializer):
         model = User
-        fields = ['id', 'email', 'nickname', 'first_name', 'last_name',
-                  'password', 'role_id', 'created_at', 'updated_at', 'status']
+        fields = ['id', 'email', 'status',
```
keep in one line

vvelizariy commented on codeArena/user/tests.py:
```diff
@@ -1,3 +1,209 @@
+
+from rest_framework.test import APITestCase
+
+# This is the responses library mentioned in the article, which
```
copy-paste?)

vvelizariy commented on codeArena/user/tests.py:
```diff
@@ -1,3 +1,209 @@
+SOCIAL_URL = "/api/2.0/social/{}/"  # Customize for your own endpoint URL scheme
+
+
+class TryToken:  # using the python3 idiom
```
copy-paste)

vvelizariy commented on codeArena/user/tests.py:
```diff
@@ -1,3 +1,209 @@
+    Test that PSA is correctly limiting login attempts to social auth providers
+    which we have specifically enabled.
+    """
+    # obviously, if we support yahoo logins in the future,
```
one more copy-paste)

vvelizariy commented on codeArena/user/tests.py:
```diff
@@ -1,3 +1,209 @@
+# provides RequestsMock
+import responses
+
+from requests.exceptions import HTTPError
```
please format imports according PEP8

vvelizariy commented on codeArena/user/urls.py:
```diff
@@ -1,18 +1,19 @@
-    path('create_user/', views.CreateUserAPIView.as_view()),
-    path('obtain_token/', views.authenticate_user),
+    # path('create_user/', views.CreateUserAPIView.as_view()),
+    # path('obtain_token/', views.authenticate_user),
```
You should remove commented lines

okslisovska commented on codeArena/user/urls.py:
```diff
@@ -1,18 +1,19 @@
+from django import urls
```
The imported `urls` seems not to be used.

okslisovska commented on codeArena/user/urls.py:
```diff
@@ -1,18 +1,19 @@
+from django import urls
 from django.urls import path
+from django.urls.conf import include
```
The same for `include`, it is imported but not used

dmytrosmetanin committed:
Delete tests.py

Added by mistake.
Test was given with the OAuth tutorial and for testing purposes of lib authors, be4 the front-end will be written. It's unnecessary for the proj.
